slow-restart/main.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	flag "github.com/spf13/pflag"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	timeout = flag.DurationP("timeout", "t", 10*time.Minute, "how long wait for deleting pods")
+	sleep   = flag.DurationP("sleep", "s", time.Minute, "how log wait between each deployment restart")
+)
+
+func main() {
+	flag.Parse()
+
+	ctx := context.Background()
+
+	cli, err := getKubernetesCli()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	version, err := getCPlaneLinkerdVersion(ctx, cli)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	podsPerDep, err := getPodsPerDep(ctx, cli, version)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if cont, err := shouldContinue(podsPerDep); err != nil {
+		log.Fatal(err)
+	} else if !cont {
+		log.Printf("Nothing to do")
+		return
+	}
+
+	for dep, podNames := range podsPerDep {
+		if err := restartDep(ctx, dep); err != nil {
+			log.Fatal(err)
+		}
+
+		if err := waitForDeletePods(ctx, podNames, *timeout); err != nil {
+			log.Fatal(err)
+		}
+
+		time.Sleep(*sleep)
+	}
+}
+
+func getKubernetesCli() (*kubernetes.Clientset, error) {
+	kubeCfgPath := filepath.Join(os.Getenv("HOME"), ".kube", "config")
```

Comment:
Maybe we should first check the `KUBECONFIG` environment variable and if it is empty fall back to `$HOME/.kube/config`?

https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/#append-home-kube-config-to-your-kubeconfig-environment-variable